ring/util_test.go

```diff
@@ -123,48 +116,50 @@ func TestWaitRingStabilityShouldReturnAsSoonAsMinStabilityIsReachedOnNoChanges(t
 		strategy:            NewDefaultReplicationStrategy(),
 	}
 
+	return ring
+}
+
+func TestWaitRingStabilityShouldReturnAsSoonAsMinStabilityIsReachedOnNoChanges(t *testing.T) {
+	t.Parallel()
+
+	const (
+		minStability = 2 * time.Second
```

Comment:
2s is not enough here, because when we assert on the actual wait time we pass a delta of 2s, see:
```
assert.InDelta(t, minStability, elapsedTime, float64(2*time.Second))
```

If we remove the call to `WaitRingStability()` at all, the test will pass anyway. I know it's not something you changed, but I would be glad if you could fix it. One way to fix it is checking the elapsed time like we do in `TestWaitRingStabilityShouldReturnOnceMinStabilityOfInstancesHasBeenReached`.

_The same comment applies to tests below._

Reply:
Yes, I noticed this as well,  I'll add the checking for minimal time. I suspect the numbers here are chosen to avoid flakyness.  
I also think that these should not be running in real time, so I'll propose a change as part of my pet project to improve testability at a later time.